sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.service.ts

```diff
@@ -108,4 +134,152 @@ export class EducationProgramOfferingControllerService {
       description: getOfferingNameAndPeriod(offering),
     }));
   }
+
+  /**
+   * Creates the offering model to be validated and saved using the DTO received.
+   * @param institutionId institution id.
+   * @param locationId location id.
+   **The authorization to validated if the location belongs to the institution
+   **is not expected to be performed by this method.
+   * @param programId program id.
+   * @param payload information to generate the model to perform the offering
+   * validation and persistence.
+   * @returns offering model to be validated and saved.
+   */
+  async mapOfferingAPIInDTOTOSaveOfferingModel(
```

Comment:
IMO we can avoid using DTO names in method name.(e.g buildOfferingModel etc.)

Reply:
I believe this method is exclusively to build the payload to create new offering right?

Reply:
I agree. I will rename it to `buildSaveOfferingModel`.

Reply:
Renamed.